django/forms/boundfield.py

```diff
@@ -219,12 +219,28 @@ def initial(self):
         return data
 
     def build_widget_attrs(self, attrs, widget=None):
-        widget = widget or self.field.widget
         attrs = dict(attrs)  # Copy attrs to avoid modifying the argument.
-        if widget.use_required_attribute(self.initial) and self.field.required and self.form.use_required_attribute:
-            attrs['required'] = True
+        if isinstance(widget, MultiWidget) or isinstance(self.field.widget, MultiWidget):
+            widgets = widget.widgets or self.field.widget.widgets
+            for field, widget in zip(self.field.fields, widgets):
+                if (
+                    widget.use_required_attribute(self.initial) and
+                    field.required and
+                    self.form.use_required_attribute
+                ):
+                    widget.attrs["required"] = True
```

Comment:
Will this permanently adjust the attr on the widget instance?

Reply:
Yes, this makes me sad 👎 

Having looked at this again (seem to learn something new every day 👍 ) I _think_ I've been 'abusing' the wrong widget in fields.py (next comment) and I'm therefore correcting my error here.